src/libraries/System.Net.Security/src/System/Net/NegotiateAuthenticationPal.Unix.cs

```diff
@@ -764,5 +769,18 @@ internal static NegotiateAuthenticationStatusCode GetErrorCode(Interop.NetSecuri
                 }
             }
         }
+
+        public static bool CheckHasSystemNetSecurityNative()
+        {
+            try
+            {
+                return Interop.NetSecurityNative.IsNtlmInstalled();
+            }
+            catch (Exception e) when (e is EntryPointNotFoundException || e is DllNotFoundException || e is TypeInitializationException)
```

Comment:
Do we really need this check? Would it be better for the callers to use `!UseManagedNtlm` instead?

Reply:
Managed NTLM can sometimes use Kerberos which is implemented only in gssapi. I didn't want to disable that entirely cause there may be scenarios were managed NTLM impl is used even though the native libs are there.

https://github.com/dotnet/runtime/blob/961257bf1b28c1cc3c111a84644f029481f958f0/src/libraries/System.Net.Security/src/System/Net/NegotiateAuthenticationPal.ManagedSpnego.cs#L130-L138

https://github.com/dotnet/runtime/blob/6b21cb12b5177d56053bb30073c326dba268e494/src/libraries/System.Net.Security/src/System/Net/NegotiateAuthenticationPal.Unix.cs#L32-L42